### PR TITLE
[dashboard]: only display delete button if not preparing

### PR DESCRIPTION
**Pull request by MrSimonEmms**
Proposed fix for #4173

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **svenefftinge** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @svenefftinge` in a comment when ready.

Associated issue: *#4173*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by gtsiolis** [comment]
Looking at this now! 👀

**Review by gtsiolis — COMMENTED** [review]
Lunch happened but I'm back! 🍝 

Changes look good, @MrSimonEmms!

Ideally, we could allow users to click the delete action and handle all the necessary checks on the backstage even if that's moving the to-be-deleted entry to outside the active workspaces list, as described in option :b: in https://github.com/gitpod-io/gitpod/issues/4173#issuecomment-874248212. 🌤️ 

However, I understand there's a technical limitation to do this (https://github.com/gitpod-io/gitpod/issues/4173#issuecomment-884013035) so removing the delete option sounds like a good MVC (minimum viable change) especially since this removal is visible at rare times as the user usually switches context to starting a workspace (loading screen) away from the dashboard workspaces list. ✔️ 

I've added one small nitpick issue about borders but could be fine to skip this, too. Let me know what you think.

File: components/dashboard/src/workspaces/WorkspaceEntry.tsx [hunk 1]
```diff
@@ -74,14 +74,18 @@ export function WorkspaceEntry({ desc, model, isAdmin, stopWorkspace }: Props) {
...
             },
-            {
+        );
+
+        // Workspaces can only be deleted if not preparing
+        if (state !== 'preparing') {
+            menuEntries.push({
                 title: 'Delete',
                 customFontStyle: 'text-red-600 dark:text-red-400 hover:text-red-800 dark:hover:text-red-300',
                 onClick: () => {
```
[the remark this answers is not in the document]
Comment (MrSimonEmms) [reply components/dashboard/src/workspaces/WorkspaceEntry.tsx:84]: Yeah, that should be a fairly easy issue to fix - I'll just make the separator conditional.

**Comment by gtsiolis** [comment]
**issue**: @MrSimonEmms For some reason I canno creaste any new workspace, getting the following two errors:

```
cannot pull image: rpc error: code = Unknown desc = failed to pull and unpack image "reg.sje-cannot-delete-ws.staging.gitpod-dev.com:30474/remote/d0b360df-d75f-496a-a3ef-1b8a19cee9bb:latest": failed to resolve reference "reg.sje-cannot-delete-ws.staging.gitpod-dev.com:30474/remote/d0b360df-d75f-496a-a3ef-1b8a19cee9bb:latest": failed to do request: Head https://reg.sje-cannot-delete-ws.staging.gitpod-dev.com:30474/v2/remote/d0b360df-d75f-496a-a3ef-1b8a19cee9bb/manifests/latest: dial tcp 127.0.0.1:30474: connect: connection refused
```

```
cannot pull image: Back-off pulling image "reg.sje-cannot-delete-ws.staging.gitpod-dev.com:30474/remote/bcce5a7d-da3a-431d-8ebc-0b83386bace9"
```

Maybe there's an issue with preview environments at the moment? 🤷

**Comment by corneliusludmann** [comment]
Due to https://github.com/gitpod-io/gitpod/issues/4173#issuecomment-872968251:

> Deleting workspaces in any phase but RUNNING or STOPPED is very difficult. I reckon we should rather disable the "Delete" item in the menu if the workspace is neither running nor stopped.

What about checking if the workspace is in RUNNING or STOPPED instead and only provide the delete button during these 2 phases?

→ This would probably fix this as well: https://community.gitpod.io/t/unable-to-delete-workspace/4956

**Pull request closed without merging**